presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeAdjacentWindows.java

```diff
@@ -60,4 +61,16 @@
                 parent.getPrePartitionedInputs(),
                 parent.getPreSortedOrderPrefix()));
     }
+
+    private static boolean dependsOn(WindowNode parent, WindowNode child)
```

Comment:
@maciejgrzybek I have copied this method from your branch. If you can move this to `WindowNode` class, then I can use the common code.